service/service.go

```diff
@@ -190,15 +210,18 @@ func (app *Application) runAndWaitForShutdownEvent() {
 	}
 }
 
-func (app *Application) setupConfigurationComponents() {
+func (app *Application) setupConfigurationComponents(cFactory ConfigFactory) {
 	// Load configuration.
 	app.logger.Info("Loading configuration...")
-	cfg, err := config.Load(app.v, app.factories, app.logger)
+	cfg, err := cFactory(app.v, app.factories)
 	if err != nil {
 		log.Fatalf("Cannot load configuration: %v", err)
 	}
-
 	app.config = cfg
```

Comment:
nit: you may want to do the assignment after validation?
(doesn't matter now because of `log.Fatal`, but may change later due to #619)

Reply:
I have moved it after the validation